.github/composites/docker-build-push/action.yaml

```diff
@@ -0,0 +1,115 @@
+name: 'Build and Publish Docker Image'
+description: 'Build and Publish Docker Image'
+inputs:
+  repository:
+    description: 'Docker hub repository'
+    required: true
+    default: 'cloudcustodian'
+  name:
+    description: 'image name, should match dockerfile name'
+    required: true
+  push:
+    description: 'Push image: true/false'
+    required: true
+    default: false
+  poetry_version:
+    description: "Poetry Version to use"
+    default: "1.1.14"
+  trivy_version:
+    description: "trivy version to use"
+    default: "0.5.4"
+  platforms:
+    description: "Platforms to build, e.g. linux/arm64,linux/amd64"
+    default: "linux/amd64"
+runs:
+  using: "composite"
+  steps:
+    - name: Set up QEMU
+      uses: docker/setup-qemu-action@v1
+    - name: Set up Docker Buildx
+      id: buildx
+      uses: docker/setup-buildx-action@v1
+    - name: Set up Python 3.10
```

Comment:
let's move this closer to its usage (test) action, and utilize the poetry cache it supports
https://github.com/actions/setup-python

Reply:
Just realized alternatively we use the same cache setup we do for regular co so it’s gets shared.

Reply:
well we test on 3.9 and this runs on 3.10 do you want to move the ci up to 3.10 as well?